### ISSUE TITLE
Cannot rename public types

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#6482]
```
ERROR [2018-10-26 18:30:44Z]: Error while renaming symbol ReplaceWithSingleCallToLastTests
System.InvalidOperationException: There is no atomic undo operation running.
  at Mono.TextEditor.TextDocument.EndAtomicUndo () [0x000bc] in /Users/vsts/agent/2.141.1/work/1/s/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/Document/TextDocument.cs:1226 
  at Mono.TextEditor.TextDocument+UndoGroup.Dispose () [0x00008] in /Users/vsts/agent/2.141.1/work/1/s/monodevelop/main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/Document/TextDocument.cs:1175 
  at MonoDevelop.Refactoring.TextReplaceChange+<>c.<FinishRefactoringOperation>b__22_0 (System.IDisposable grp) [0x00000] in /Users/vsts/agent/2.141.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring/Change.cs:92 
  at System.Collections.Generic.List`1[T].ForEach (System.Action`1[T] action) [0x0001e] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/Common/src/CoreLib/System/Collections/Generic/List.cs:578 
  at MonoDevelop.Refactoring.TextReplaceChange.FinishRefactoringOperation () [0x0000a] in /Users/vsts/agent/2.141.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring/Change.cs:92 
  at MonoDevelop.Refactoring.RefactoringService.AcceptChanges (MonoDevelop.Core.ProgressMonitor monitor, System.Collections.Generic.IList`1[T] changes, MonoDevelop.Ide.ITextFileProvider fileProvider) [0x0024f] in /Users/vsts/agent/2.141.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring/RefactoringService.cs:191 
  at MonoDevelop.Refactoring.RefactoringService.AcceptChanges (MonoDevelop.Core.ProgressMonitor monitor, System.Collections.Generic.IList`1[T] changes) [0x00000] in /Users/vsts/agent/2.141.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring/RefactoringService.cs:125 
  at MonoDevelop.Refactoring.Rename.RenameItemDialog.OnOKClicked (System.Object sender, System.EventArgs e) [0x001b9] in /Users/vsts/agent/2.141.1/work/1/s/monodevelop/main/src/addins/MonoDevelop.Refactoring/MonoDevelop.Refactoring.Rename/RenameItemDialog.cs:245 
```

> VS bug [#718076](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/718076)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.1167 (master)

Pull Request #6490 merged by: Lluis Sanchez
Author: mono
Commit: 0a1dbb8386e4c7958669378449e8be4fa34d6659 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.1840 (release-7.7)

Pull Request #6585 merged by: Lluis Sanchez
Author: mono
Commit: 088e95781f983bdbc3091b3260ada57925b198f7 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.1193 (release-7.8)

Pull Request #6490 merged by: Lluis Sanchez
Author: mono
Commit: 0a1dbb8386e4c7958669378449e8be4fa34d6659 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.658 (release-8.0)

Pull Request #6490 merged by: Lluis Sanchez
Author: mono
Commit: 0a1dbb8386e4c7958669378449e8be4fa34d6659 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #6490 merged by: Lluis Sanchez
Author: mono
Commit: 0a1dbb8386e4c7958669378449e8be4fa34d6659 (xamarin/monodevelop)
